### PR TITLE
Remove telemetry toolkit component

### PR DESCRIPTION
**Pull request by softwaretechone**
Merge pull request #1 from softwaretechone/master merges pull request #1281 into this pull request.

Part 14 makes the corrections as stated in the review of pull request #1281 

Part 15 deletes the mailnews telemetry.

### CODE REVIEW

**Comment by wolfbeast** [comment]
Please check buildability, seems you forgot to remove the task cleanup with the corrections made.
```
 7:14.23 e:/mozdev/UXP/dom/media/platforms/wmf/WMFAudioMFTManager.cpp(252): error C2065: 'task': und
eclared identifier
 7:14.23 e:/mozdev/UXP/dom/media/platforms/wmf/WMFAudioMFTManager.cpp(252): error C2228: left of '.f
orget' must have class/struct/union
 7:14.23 e:/mozdev/UXP/dom/media/platforms/wmf/WMFAudioMFTManager.cpp(252): note: type is 'unknown-t
ype'
 7:14.23 e:/mozdev/UXP/dom/media/platforms/wmf/WMFVideoMFTManager.cpp(119): error C2065: 'task': und
eclared identifier
 7:14.23 e:/mozdev/UXP/dom/media/platforms/wmf/WMFVideoMFTManager.cpp(119): error C2228: left of '.f
orget' must have class/struct/union
 7:14.24 e:/mozdev/UXP/dom/media/platforms/wmf/WMFVideoMFTManager.cpp(119): note: type is 'unknown-t
ype'
```

**Comment by mattatobin** [comment]
Would you be willing to also do a pull request for binoc-central removing the telemetry for mail and navigator?

**Comment by softwaretechone** [comment]
I issued the fix. Yes, I am willing to do that.

**Comment by wolfbeast** [comment]
> Your commit messages are also STILL incorrect and badly described as well.

Despite this I'm willing to merge in this case since I really don't feel like redoing the commits myself for their meta. Future PRs will be rejected because of it, though, since we do need a clear, searchable repo history where possible.

**Comment by g4jc** [comment]
As written recommend **r-**. [Basilisk] is completely busted.

`
*************************
A coding exception was thrown and uncaught in a Task.

Full message: TypeError: Services.telemetry is undefined
Full stack: _updateTelemetry@resource://gre/modules/osfile/osfile_async_front.jsm:459:9
post/<@resource://gre/modules/osfile/osfile_async_front.jsm:438:11
TaskImpl_run@resource://gre/modules/Task.jsm:319:42
process@resource://gre/modules/Promise.jsm -> resource://gre/modules/Promise-backend.js:932:23
walkerLoop@resource://gre/modules/Promise.jsm -> resource://gre/modules/Promise-backend.js:813:7
scheduleWalkerLoop/<@resource://gre/modules/Promise.jsm -> resource://gre/modules/Promise-backend.js:747:11

*************************

chrome://browser/content/browser-trackingprotection.js, line 63: TypeError: Services.telemetry is undefined`

I also believe that oneOff search function which was purged in Part 10 was not properly fixed in Part 14. I will see if this is fixable and can rebase the patches with proper commit titles. Will advise.

**Comment by mattatobin** [comment]
Did you even test to see if this builds @softwaretechone. I am getting very weary of this.

**Comment by adeshkp** [comment]
If he manages to fix it, then fine. Otherwise I'm willing to kill it (slowly, in multiple parts) as I find time to work on this. I've already removed a sufficiently big chunk of telemetry from the code. I know what to do and how to do it.

**Comment by g4jc** [comment]
Spent a few hours on this, here's the situation:

* All histograms were missed. This breaks the global toolkit/components affecting all applications.

* Parts of Pale Moon telemetry were missed

* Basically all of Basilisk telemetry components which were overlooked. It appears none of the `Services.telemetry` JS Functions were even considered. PdfJsTelemetry, BrowserUsageTelemetry, etc.

* All XUL elements / CSS related to telemetry preferences are still there

*  Various core functions and usability is currently broken due to toolkit bustage.

In short, it's missing about 20 more commits to make this run properly - but with so many different components busted I'm not sure where to start. I got it building with this [hack of a patch](https://pastebin.com/raw/hLKAqBwm) and it's still partially busted due to telemetry time stamps missing and not properly removed.

I think we should probably do this the right way instead of risking the stability of all applications by taking a chainsaw to it.

**Comment by mattatobin** [comment]
I have to agree with @g4jc

r-

**Comment by wolfbeast** [comment]
Considering the feedback, clearly unacceptable. Closing.
